Vastly improved error messages from the API

bcampeau opened the pull request:
Fixes #218

mslinnea commented on admin/class-admin-apple-notice.php:
```diff
@@ -134,6 +134,22 @@ public static function show() {
+			// If it's not 2, it's too unclear how to proceed.
+			// Try to split the second param on commas
+			$errors = explode( ',', $message_array[1] );
+			if ( count( $errors > 1 ) ) {
```
I think you mean `if ( count( $errors ) > 1 ) {`

mslinnea commented:
just one thing, then 🚅

bcampeau commented:
Fixed logic error from the initial review, updated to fix one more messaging issue and add release notes and a version bump for the official 1.1.6 release on Monday.

jameswburke commented:
Looks good :bee:
